unit_test/solver/matrix/test_matrix.py

```diff
@@ -0,0 +1,206 @@
+#  Copyright (c) 2023. Deltares & TNO
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Test matrix object."""
+
+import unittest
+
+import numpy as np
+
+from simulator_core.solver.matrix.matrix import Matrix
+from simulator_core.solver.matrix.equation_object import EquationObject
+
+
+class MatrixTest(unittest.TestCase):
+    """Test the matrix object."""
+
+    def test_init(self):
+        """Test the init of the matrix object."""
+        # arrange
+
+        # act
+        matrix = Matrix()
+
+        # assert
+        self.assertEqual(matrix.num_unknowns, 0)
+        self.assertEqual(len(matrix.mat), 0)
+        self.assertEqual(len(matrix.rhs), 0)
+        self.assertEqual(len(matrix.sol_new), 0)
+        self.assertEqual(len(matrix.sol_old), 0)
+        self.assertEqual(matrix.relative_convergence, 1e-6)
+        self.assertEqual(matrix.absolute_convergence, 1e-6)
+
+    def test_add_unknowns(self):
+        """Test the add unknowns of the matrix object."""
+        # arrange
+        matrix = Matrix()
+
+        # act
+        index = matrix.add_unknowns(2)
```

Comment:
voeg bij dit soort dingen even de key toe die je gebruikt. Als er nu ooit iets wijzigt weet je niet waar je naar verwijst.

Reply:
Moeten we hier niet checken dat number_unknowns > 0?

Reply:
Toegevoegd